mne/viz/_3d.py

```diff
@@ -1460,31 +1460,47 @@ def _plot_mpl_stc(stc, subject=None, surface='inflated', hemi='lh',
                   colormap='auto', time_label='auto', smoothing_steps=10,
                   subjects_dir=None, views='lat', clim='auto', figure=None,
                   initial_time=None, time_unit='s', background='black',
-                  spacing='oct6', time_viewer=False):
+                  spacing='oct6', time_viewer=False, colorbar=True,
+                  transparent=True):
     """Plot source estimate using mpl."""
     import matplotlib.pyplot as plt
     from mpl_toolkits.mplot3d import Axes3D
     from matplotlib import cm
+    from matplotlib import colors
     from matplotlib.widgets import Slider
     import nibabel as nib
     from scipy import sparse, stats
     from ..morph import _get_subject_sphere_tris
     if hemi not in ['lh', 'rh']:
         raise ValueError("hemi must be 'lh' or 'rh' when using matplotlib. "
                          "Got %s." % hemi)
-    kwargs = {'lat': {'elev': 5, 'azim': 0},
-              'med': {'elev': 5, 'azim': 180},
-              'fos': {'elev': 5, 'azim': 90},
-              'cau': {'elev': 5, 'azim': -90},
-              'dor': {'elev': 90, 'azim': 0},
-              'ven': {'elev': -90, 'azim': 0},
-              'fro': {'elev': 5, 'azim': 110},
-              'par': {'elev': 5, 'azim': -110}}
-    if views not in kwargs:
-        raise ValueError("views must be one of ['lat', 'med', 'fos', 'cau', "
+    lh_kwargs = {'lat': {'elev': 0, 'azim': 180},
+                 'med': {'elev': 0, 'azim': 0},
+                 'ros': {'elev': 0, 'azim': 90},
+                 'cau': {'elev': 0, 'azim': -90},
+                 'dor': {'elev': 90, 'azim': -90},
+                 'ven': {'elev': -90, 'azim': -90},
+                 'fro': {'elev': 0, 'azim': 106.739},
+                 'par': {'elev': 30, 'azim': -120}}
+    rh_kwargs = {'lat': {'elev': 0, 'azim': 0},
+                 'med': {'elev': 0, 'azim': 180},
+                 'ros': {'elev': 0, 'azim': 90},
+                 'cau': {'elev': 0, 'azim': -90},
+                 'dor': {'elev': 90, 'azim': -90},
+                 'ven': {'elev': -90, 'azim': -90},
+                 'fro': {'elev': 16.739, 'azim': 60},
+                 'par': {'elev': 30, 'azim': -60}}
+    kwargs = dict(lh=lh_kwargs, rh=rh_kwargs)
+    if views not in lh_kwargs:
+        raise ValueError("views must be one of ['lat', 'med', 'ros', 'cau', "
                          "'dor' 'ven', 'fro', 'par']. Got %s." % views)
     ctrl_pts, colormap, _, _ = _limits_to_control_points(
-        clim, stc.data, colormap, transparent=False, fmt='matplotlib')
+        clim, stc.data, colormap, transparent=transparent, fmt='matplotlib')
+    colormap = plt.get_cmap(colormap)
+
+    locs = np.clip(np.linspace(ctrl_pts[0] / ctrl_pts[2], 1, 256), 0, 1)
+    colormap = colormap(locs)
+    colormap = colors.ListedColormap(colormap)
```

Comment:
You also should not need to do this, the `_limits_to_control_points` should already do the correct nonlinear scaling

Reply:
I tried the following:

```py

    ctrl_norm = ctrl_pts - ctrl_pts[0]
    ctrl_norm /= ctrl_norm[2]
    vals = np.interp(np.linspace(0, 1, 256), ctrl_norm, [0., 0.5, 1.])
    colormap = np.array(colormap(vals))
    if transparent:
        colormap[:, 3] = np.interp(np.linspace(0, 1, 256), ctrl_norm,
                                   [0, 1, 1])
    colormap = colors.ListedColormap(colormap)
```

but for some strange reason, the brainmap starts saturating then:

![brainmap](https://user-images.githubusercontent.com/15852194/46231932-9a0f9780-c33b-11e8-9582-474ca40f4e55.png)

I can't seem to get both the colorbar and brainmap to look good ...